### PR TITLE
feat(base-cluster/grafana): add persistence and configuration options…

### PR DESCRIPTION
**Pull request by tasches**
… to grafana

### CODE REVIEW

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml [hunk 1]
```diff
@@ -13,6 +13,11 @@ securityContext:
     type: RuntimeDefault
 containerSecurityContext: {{- include "base-cluster.prometheus-stack.containerSecurityContext" (dict) | nindent 2 }}
 resources: {{- include "common.resources" .Values.monitoring.grafana | nindent 2 }}
+{{- if .Values.monitoring.grafana.persistence.enabled }}
+persistence:
```
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml:17]: Missing storageClass, see https://github.com/SyeKlu/teutonet-helm-charts/blob/main/charts/base-cluster/templates/nfs-server-provisioner/nfs-server-provisioner.yaml#L22-L25

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml [hunk 2]
```diff
@@ -146,28 +151,11 @@ envValueFrom:
...
-    scopes: openid profile email
-    {{- end }}
-  {{- end }}
-  server:
-    root_url: {{ printf "https://%s" (include "base-cluster.grafana.host" .) }}
+{{- if $.Values.monitoring.grafana.config -}}
+  {{ merge $.Values.monitoring.grafana.config (include "base-cluster.grafana.config" $ | fromYaml) | toYaml | nindent 4 -}}
+{{- else -}}
+  {{ include "base-cluster.grafana.config" $ | nindent 4 -}}
+{{- end }}
```
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml:158]:
```suggestion
  {{ merge ($.Values.monitoring.grafana.config | default (dict)) (include "base-cluster.grafana.config" $ | fromYaml) | toYaml | nindent 4 -}}
```

File: charts/base-cluster/values.schema.json
Comment (cwrau) [inline charts/base-cluster/values.schema.json:1]: I think you're missing the schema for the `config` block?

**Review by cwrau — CHANGES_REQUESTED** [review]

File: charts/base-cluster/values.schema.json
Comment (tasches) [reply charts/base-cluster/values.schema.json:1]: it is already in the chart

Comment (tasches) [reply charts/base-cluster/values.schema.json:1]: https://github.com/teutonet/teutonet-helm-charts/blob/main/charts/base-cluster/values.schema.json#L746

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml (hunk 1 above)
Comment (tasches) [reply charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml:17]: Oh I was somehow blind :grin: I looked at the grafana helm [values](https://github.com/grafana/helm-charts/blob/main/charts/grafana/values.yaml#L367-L383) and didn't notice the comment with the storageclassname :facepalm: .

File: charts/base-cluster/values.schema.json
Comment (cwrau) [reply charts/base-cluster/values.schema.json:1]: Oh, didn't know that.

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_helpers.yaml [hunk 3]
```diff
@@ -18,6 +18,31 @@ ingress:
...
+      client_secret: ${OIDC_CLIENT_SECRET}
+      name: OAuth
+      role_attribute_path: {{ .grafana.roleAttributePath | quote }}
+      scopes: openid profile email
+    {{- end }}
+  {{- end }}
+    server:
+      root_url: {{ printf "https://%s" (include "base-cluster.grafana.host" .) }}
+{{- end -}}
+
```
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_helpers.yaml:45]: This should rather be in the `_grafana-config.yaml`

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml [hunk 4]
```diff
@@ -13,6 +13,12 @@ securityContext:
     type: RuntimeDefault
 containerSecurityContext: {{- include "base-cluster.prometheus-stack.containerSecurityContext" (dict) | nindent 2 }}
 resources: {{- include "common.resources" .Values.monitoring.grafana | nindent 2 }}
+{{- if .Values.monitoring.grafana.persistence.enabled }}
+persistence:
+  enabled: true
+  size: {{ .Values.monitoring.grafana.persistence.size | default "10Gi" }}
+  storageClassName: {{ .Values.monitoring.grafana.persistence.storageClassName | default "default" }}
```
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml:20]: See https://github.com/teutonet/teutonet-helm-charts/blob/main/charts/base-cluster/templates/monitoring/loki/loki.yaml#L36

**Review by cwrau — CHANGES_REQUESTED** [review]

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_helpers.yaml (hunk 3 above)
Comment (tasches) [reply charts/base-cluster/templates/monitoring/kube-prometheus-stack/_helpers.yaml]: moved

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml (hunk 4 above)
Comment (tasches) [reply charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml]: please take a look at the latest changes

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml [hunk 5]
```diff
@@ -1,3 +1,28 @@
+{{- define "base-cluster.grafana.config" -}}
```
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml:1]:
```suggestion
{{- define "base-cluster.prometheus-stack.grafana.ini" -}}
```

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml [hunk 6]
```diff
@@ -146,28 +178,7 @@ envValueFrom:
...
-    client_id: {{ .config.clientId }}
-    client_secret: ${OIDC_CLIENT_SECRET}
-    name: OAuth
-    role_attribute_path: {{ .grafana.roleAttributePath | quote }}
-    scopes: openid profile email
-    {{- end }}
-  {{- end }}
-  server:
-    root_url: {{ printf "https://%s" (include "base-cluster.grafana.host" .) }}
+  {{ merge ($.Values.monitoring.grafana.config | default (dict)) (include "base-cluster.grafana.config" $ | fromYaml) | toYaml | nindent 4 -}}
```
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml:181]:
```suggestion
  {{ merge ($.Values.monitoring.grafana.config | default (dict)) (include "base-cluster.prometheus-stack.grafana.ini" $ | fromYaml) | toYaml | nindent 4 -}}
```

**Review by cwrau — CHANGES_REQUESTED** [review]

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml (hunk 4 above)
Comment (cwrau) [reply charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml]: Still valid

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml [hunk 7]
```diff
@@ -1,3 +1,32 @@
+{{- define "base-cluster.prometheus-stack.grafana.ini.ingress" -}}
+  {{- $host := printf "https://%s" (include "base-cluster.grafana.host" .context) -}}
+auth:
+  signout_redirect_url: {{ $host }}
+  server:
+    root_url: {{ $host }}
```
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_grafana-config.yaml:6]: Ups;

```suggestion
server:
  root_url: {{ $host }}
auth:
  signout_redirect_url: {{ $host }}
```

**Review by cwrau — APPROVED** [review]

**Pull request merged**